tests/test_cli.py

```diff
@@ -330,3 +344,29 @@ def test_download_a_file(tmp_path, patched_repo_and_user, repo):
     os.chdir(working_dir)
     main(["dl", "hello.md"])
     assert (working_dir / "hello.md").read_bytes() == b"awesome content"
+
+
+def test_download_a_file_from_public_repo(
+    tmp_path, patched_another_repo_and_user, outside_repo, requests_mock, patched_repo_and_user
+):
+    requests_mock.get(
+        "https://raw.githubusercontent.com/another_awesome/repository/main/hello.md", content=b"awesome content"
+    )
+    working_dir = tmp_path / "working_dir"
+    working_dir.mkdir()
+    os.chdir(working_dir)
+    main(["dl", "https://github.com/another_awesome/repository/blob/main/hello.md"])
+    assert (working_dir / "hello.md").read_bytes() == b"awesome content"
+
+
+def test_download_a_file_from_public_repo_rasie_error(
+    tmp_path, patched_another_repo_and_user, outside_repo, requests_mock, patched_repo_and_user
+):
+    requests_mock.get("https://raw.githubusercontent.com/another_awesome/repository/main/hello.md", status_code=400)
+    working_dir = tmp_path / "working_dir"
+    working_dir.mkdir()
+    os.chdir(working_dir)
+    main(["dl", "https://github.com/another_awesome/repository/blob/main/hello.md"])
+    with pytest.raises(Exception) as exc_info:
+        raise Exception("There was a problem with your download, please check url")
+    assert str(exc_info.value) == "There was a problem with your download, please check url"
```

Comment:
this has no sense Alvar. It's a self-fulfilling prophecy: "I will prove that the following code raise an exception " and the following code is... literally `raise Exception()`.